File: Java/1957.delete-characters-to-make-fancy-string.java
/*
 * @lc app=leetcode id=1957 lang=java
 *
 * [1957] Delete Characters to Make Fancy String
 */

// @lc code=start
class Solution {

    public String makeFancyString(String s) {
        if (s.isEmpty()) {
            return "";
        }

        StringBuilder builder = new StringBuilder();

        int count = 1;
        builder.append(s.charAt(0));

        for (int i = 1; i < s.length(); i++) {
            if (s.charAt(i) == s.charAt(i - 1)) {
                count++;
            } else {
                count = 1;
            }

            if (count <= 2) {
                builder.append(s.charAt(i));
            }
        }
        return builder.toString();
    }
}
// @lc code=end

File: Python/1957.delete-characters-to-make-fancy-string.py
#
# @lc app=leetcode id=1957 lang=python3
#
# [1957] Delete Characters to Make Fancy String
#

# @lc code=start
class Solution:
    def makeFancyString(self, s: str) -> str:
        if not s:
            return ""
        
        res = ""
        count = 1
        res += s[0]
        for i in range(1, len(s)):
            if s[i] == s[i - 1]:
                count += 1
            else:
                count = 1
            
            if count <= 2:
                res += s[i]
        return res
# @lc code=end

